util/dataset.go

```diff
@@ -147,17 +150,55 @@ func (d *DataSet) Validate() error {
 	}
 
 	if d.IngestPipeline == "" && len(paths) > 0 {
-		return fmt.Errorf("Package contains pipelines which are not used: %v, %s", paths, d.ID)
+		return fmt.Errorf("unused pipelines in the package (dataSetID: %s): %s", d.ID, strings.Join(paths, ","))
 	}
 
 	// In case an ingest pipeline is set, check if it is around
 	if d.IngestPipeline != "" {
-		_, errJSON := os.Stat(filepath.Join(pipelineDir, d.IngestPipeline+".json"))
-		_, errYAML := os.Stat(filepath.Join(pipelineDir, d.IngestPipeline+".yml"))
+		jsonPipelinePath := filepath.Join(pipelineDir, d.IngestPipeline+".json")
+		_, errJSON := os.Stat(jsonPipelinePath)
 
-		if os.IsNotExist(errYAML) && os.IsNotExist(errJSON) {
-			return fmt.Errorf("Defined ingest_pipeline does not exist: %s", pipelineDir+d.IngestPipeline)
+		yamlPipelinePath := filepath.Join(pipelineDir, d.IngestPipeline+".yml")
+		_, errYAML := os.Stat(yamlPipelinePath)
+
+		var validFound bool
+		if !os.IsNotExist(errJSON) {
+			err = validateIngestPipelineFile(jsonPipelinePath)
+			if err != nil {
+				return errors.Wrapf(err, "validating ingest pipeline JSON file failed (path: %s)", jsonPipelinePath)
+			}
+			validFound = true
+		}
+		if !os.IsNotExist(errYAML) {
+			err = validateIngestPipelineFile(yamlPipelinePath)
+			if err != nil {
+				return errors.Wrapf(err, "validating ingest pipeline YAML file failed (path: %s)", jsonPipelinePath)
+			}
+			validFound = true
+		}
+
+		if !validFound {
+			return fmt.Errorf("defined ingest_pipeline does not exist: %s", pipelineDir+d.IngestPipeline)
 		}
 	}
 	return nil
 }
+
+func validateIngestPipelineFile(pipelinePath string) error {
+	f, err := ioutil.ReadFile(pipelinePath)
+	if err != nil {
+		return errors.Wrapf(err, "reading ingest pipeline file failed")
```

Comment:
It would be nice to have the path reported as part of the error which makes it much easier to find which file did not parse.

Reply:
Fixed